File: 1주차/김도형/[1차] 캐시.py
# Programmers
from collections import deque
def solution(cacheSize, cities):
    if cacheSize == 0 :
        return 5 * len(cities)
    
    cities = [i.lower() for i in cities]
    
    data = deque([])
    ans = 0
    for x in cities :     
        x = x.lower()
        if data and x in data :
            ans += 1
            data.remove(x)
            data.appendleft(x)
        else :
            if len(data) >= cacheSize :
                if data :
                    data.pop()
                data.appendleft(x)
            else :
                data.appendleft(x)
            ans += 5
    
    return ans

File: 43주차/이수민/[1차] 캐시.java
import java.util.*;
class Solution {
  
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        ArrayList<String> list = new ArrayList<>();
        if(cacheSize == 0 ){
            return cities.length *5;
        }
        
        for(int i=0;i<cities.length;i++){
            String s = cities[i].toLowerCase();
            if(list.size() == cacheSize){
               if(list.contains(s)){
                   list.remove(s);
                   list.add(s);
                   answer+=1;
                   continue;
               }else{
                   list.remove(0);
               }
            }else{
                if(list.contains(s)){
                    list.remove(s);
                    answer+=1;
                    list.add(s);
                    continue;
                }
            }
            list.add(s);
            answer+=5;
        }

        return answer;
    }
}
